src/Chainweb/Mempool/InMem.hs

```diff
@@ -348,6 +333,68 @@ insertCheckInMem cfg lock txs
     hasher :: t -> TransactionHash
     hasher = txHasher (_inmemTxCfg cfg)
 
+-- |
+--
+--   This function is used when a transaction is inserted into the mempool. It is
+--   NOT used when a new block is created. For the latter more strict validation
+--   methods are used. In particular TTL validation is uses the current time as
+--   reference in the former case and the creation time of the parent header in
+--   the latter case.
```

Comment:
There are some more specific comments we can make, reflecting that this is only done for transactions coming in on the service API.